poclu/misc.c

```diff
@@ -272,6 +273,18 @@ poclu_write_file (const char *filename, char *content, size_t size)
   return 0;
 }
 
+int
+poclu_write_file (const char *filename, char *content, size_t size)
```

Comment:
This name (and function) is unspecific, while the other two are quite specific WRT mode.

The `poclu_write_file_in_mode` is useful if we e.g. want to append, but keeping this function ("write with platform-specific binary/text handling") is IMO pointless and confusing. I would just keep `poclu_write_file_in_mode` and `poclu_write_binfile`, delete the `poclu_write_file`.

Reply:
I improved the documentation a bit on the `poclu_write_file()`. Not sure if we can delete the function, at least without deprecating it first, as it is an user facing function, I think. `poclu_write_file_in_mode()` can be made public when there is demand for it.

Reply:
`user facing function` -> the whole poclu library is for internal use only, mainly to avoid code duplication in tests. There are no guarantees on the API and it is not installed by `make install` (you can check there is no cmake `install` in poclu/CMakeLists.txt), so you can go ahead and delete the function.